pkg/operator/controllers/autosizednodes/autosizednodes_controller.go

```diff
@@ -90,7 +94,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, fmt.Errorf("could not fetch KubeletConfig: %w", err)
 	}
 
-	// If already exists, update the spec
+	// If already exists, update the spec and annotations
+	if config.Annotations == nil {
+		config.Annotations = defaultConfig.Annotations
+	}
+	if val, ok := config.Annotations[mcAnnotationName]; !ok || val != mcAnnotationValue {
```

Comment:
If this is already set, we shouldn't be changing it regardless of value, as it will disconnect the kubeletconfig from the already rendered machineconfig

Reply:
This is true, we understand MCO behaviour in this condition.